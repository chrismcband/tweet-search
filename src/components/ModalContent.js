import React, { PropTypes } from 'react';
// import './style/ModalContent.css';

const propTypes = {
  user: PropTypes.object.isRequired,
};

function ModalContent(props) {
  const user = props.user
  return (
    <div className="modal-display">
      <img className="banner" src={user.profile_banner_url}/>
      <div className="flex user-elements">
        <img className="avatar" src={user.profile_image_url_https.replace('normal', 'bigger')}/>
        <div className="info user-name-location">
          <h4>{user.name}</h4>
          <h5>{user.screen_name}</h5>
          <h6>{user.location}</h6>
        </div>
        <div className="info stats">
          <p className="stats-title">FRIENDS</p>
          <span>{user.friends_count}</span>
        </div>
        <div className="info stats">
          <p className="stats-title">FOLLOWERS</p>
          <span>{user.followers_count}</span>
        </div>
        <div className="info stats">
          <p className="stats-title">STATUSES</p>
          <span>{user.statuses_count}</span>
        </div>
      </div>
      <p className="description">{user.description}</p>
    </div>
  );
}

ModalContent.propTypes = propTypes;

export default ModalContent;
