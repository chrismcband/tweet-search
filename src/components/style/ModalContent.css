.modal-display{
  background-color: white;
  position: absolute;
  margin: 0 auto;
  max-width: 470px;
  left: 0;
  right: 0;
  top: 280px;
  overflow: auto;
  border-radius: 6px;
  z-index: 3;
  text-align: left;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
}
