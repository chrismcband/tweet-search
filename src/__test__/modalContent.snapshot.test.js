import renderer from 'react-test-renderer';
import React from 'react';
import ModalContent from '../components/ModalContent';

describe('ModalContent component renders user info correctly', () => {
  it('renders correctly', () => {
    const user = { name: "Michelle Obama", screen_name: "miche2007" };
    const rendered = renderer.create(
      <ModalContent user={user} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
