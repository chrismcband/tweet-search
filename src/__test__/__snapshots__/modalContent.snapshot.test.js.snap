exports[`ModalContent component renders user info correctly renders correctly 1`] = `
<div
  className="modal-display"
  onClick={[Function onClick]}>
  <img
    alt="user-banner"
    className="banner"
    src={undefined} />
  <div
    className="elements">
    <div
      className="flex user-elements">
      <img
        alt="user-avatar"
        className="avatar"
        src="https://pbs.twimg.com/profile_images/582941615741775873/Lc1Awfhp_bigger.jpg" />
      <div
        className="info user-name-location">
        <h4
          className="name">
          Michelle Obama
        </h4>
        <h5
          className="screen-name">
          miche2007
        </h5>
        <h6
          className="location">
          Washington DC
        </h6>
      </div>
      <div
        className="info stats">
        <p
          className="stats-title">
          FRIENDS
        </p>
        <span
          className="stats-figures">
          50000
        </span>
      </div>
      <div
        className="info stats">
        <p
          className="stats-title">
          FOLLOWERS
        </p>
        <span
          className="stats-figures">
          999
        </span>
      </div>
      <div
        className="info stats">
        <p
          className="stats-title">
          STATUSES
        </p>
        <span
          className="stats-figures">
          300
        </span>
      </div>
    </div>
    <p
      className="description">
      They go low, you go high
    </p>
  </div>
</div>
`;
